packages/engine-server/src/drivers/file/storev2.ts

```diff
@@ -902,10 +923,6 @@ export class FileStorage implements DStore {
     notesChangedEntries = changedFromDelete
       .concat(changeFromWrite)
       .concat(notesChangedEntries);
-    // remove duplicate updates
-    notesChangedEntries = _.uniqBy(notesChangedEntries, (ent) => {
```

Comment:
Was this causing an issue?

Reply:
yes.

in the past we just updated the existing stub notes so we didn't need multiple updates to the parent / children. but now we are explicitly deleting and creating the replacing stub in this case, we need to emit multiple events.

the `uniqBy` predicate isn't really checking if the change entries are truly duplicate. there are some properties of the entries that are not duplicates and we need to emit them in order to guarantee that plugin state is the same with the engine.

There are also some cases where we emitted delete _after_ create, which caused the tree view to omit newly created notes (because the plugin saw them as deleted). These would be corrected when we reload since the plugin catches up with whatever is persisted in the filesystem during init.

I haven't exhaustively checked for everything, but I suspect that a lot of the tree view mismatch is caused because we are not emitting the events in the right order. I'll fix them as I find them moving forward.